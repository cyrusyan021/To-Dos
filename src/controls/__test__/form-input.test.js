import React from "react"
import { shallow } from 'enzyme';

import Input from 'controls/form/input';

describe("Form - Input", () => {
  const props = {
    value: '',
    onChange: jest.fn(),
  };

  afterEach(() => {
    jest.clearAllMocks();
  });

  it("renders correctly", () => {
    const wrapper = shallow(<Input />);
    expect(wrapper).toMatchSnapshot();
  });

  it("handles value on change", () => {
    const event = {
      preventDefault: ()=>{},
      target: { value: 'change' },
    };
    const wrapper = shallow(<Input { ...props } />);
    wrapper.find('input').first().simulate('change', event);
    expect(props.onChange).toHaveBeenCalled();
    expect(props.onChange).toHaveBeenCalledWith(event);
  });
});