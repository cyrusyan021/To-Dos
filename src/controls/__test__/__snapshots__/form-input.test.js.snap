// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form - Input renders correctly 1`] = `ShallowWrapper {}`;
